src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -536,14 +541,11 @@ public class ConsoleLogsViewModel
         public required string Status { get; set; }
         public required SelectViewModel<ResourceTypeDetails> SelectedOption { get; set; }
         public required ResourceViewModel? SelectedResource { get; set; }
-        public required bool HideTimestamp { get; set; }
     }
 
-    public class ConsoleLogsPageState
-    {
-        public string? SelectedResource { get; set; }
-        public bool HideTimestamp { get; set; }
-    }
+    public record ConsoleLogsPageState(string? SelectedResource);
+
+    public record ConsoleLogTimestampState(bool HideTimestamp);
```

Comment:
Name the type so it could have other settings in the future.
```suggestion
    public record ConsoleLogConsoleSettings(bool HideTimestamp);
```